lightning/src/ln/channel.rs

```diff
@@ -4206,7 +4338,7 @@ pub(super) struct DualFundingChannelContext {
 	/// minus any fees paid for our contributed weight. This means that change will never be generated
 	/// and the maximum value possible will go towards funding the channel.
 	#[allow(dead_code)] // TODO(dual_funding): Remove once contribution to V2 channels is enabled.
-	pub our_funding_inputs: Vec<(TxIn, TransactionU16LenLimited)>,
+	pub our_funding_inputs: Option<Vec<(TxIn, TransactionU16LenLimited)>>,
```

Comment:
Will there ever be a situation where we will need these inputs again? Just wondering what the implications are for `take`'ing this.

Reply:
I think the intention with the `Option<Vec>` was so that it can be `take`n to avoid cloning.
I've changed that to simply `Vec`, and use `mem::swap` to take it out.

Reply:
Chatted a bit more offline. My concerns were around whether we are leaving ourselves in an invalid state by swapping / taking. For dual funding, IIUC, any failure in `begin_interactive_funding_tx_construction` would result in us dropping the channel (@dunxen is this accurate?). And for splicing, any failure would result in us abandoning the splice. So maybe this is ok.

However, it would be nice if our phase transitions could capture this. For instance, would it make sense to have transitions that effectively encapsulated `DualFundingChannelContext`, `InteractiveTxConstructor`, and `InteractiveTxSigningSession` in _separate_ phases akin to https://github.com/lightningdevkit/rust-lightning/pull/3423#discussion_r1857415099?

No need to hold up this PR on this as it would be better to do after #3418. But would love to hear what others think. Specifically, can we get away with forgetting about `DualFundingChannelContext` once and `InteractiveTxConstructor` is created? In #2302, I see that it is included in a funded `Channel` in an `Option`. How is it needed then?

Reply:
@jkczyz, Well for dual_funding we do go ahead and just abandon the channel even though it's technically could just be a `tx_abort` which means we could start negotiating again without abandoning the channel. However, at that stage we might need the holder to provide other inputs anyway which would need to happen on initiating or accepting a V2 channel. So best to just close it.

For splicing, yeah I'd imagine it would be abandoning the splice. @optout21 can confirm if that's what we'll do.

Reply:
Yes, in case of failed negotiation state we should abandon the splicing process. With the current solution that involves dropping the OutboundV2Channel instance -- much like in case of open -- and the already Funded original channel instance just remains.